Using Netty 4.2.7 with Java 25 on z/OS: java.lang.OutOfMemoryError: malloc(2) failed to allocate 0 bytes

We had been using Netty 4.1.126, and it works fine on Java 25.0.1 on z/OS.  However when we uplifted to 4.2.7, we start getting exceptions with the following cause.  Change any other part of the environment (earlier Java, different os) the problem goes away.  

Caused by: java.lang.OutOfMemoryError: malloc(2) failed to allocate 0 bytes
at io.netty.util.internal.CleanerJava24Linker.malloc(CleanerJava24Linker.java:214)
at io.netty.util.internal.CleanerJava24Linker$CleanableDirectBufferImpl.(CleanerJava24Linker.java:232)
at io.netty.util.internal.CleanerJava24Linker$CleanableDirectBufferImpl.(CleanerJava24Linker.java:227)
at io.netty.util.internal.CleanerJava24Linker.allocate(CleanerJava24Linker.java:198)
at io.netty.util.internal.PlatformDependent.allocateDirect(PlatformDependent.java:600)
at io.netty.buffer.EmptyByteBuf.(EmptyByteBuf.java:43)
at io.netty.buffer.AbstractByteBufAllocator.(AbstractByteBufAllocator.java:82)
at io.netty.buffer.PooledByteBufAllocator.(PooledByteBufAllocator.java:274)
at io.netty.buffer.PooledByteBufAllocator.(PooledByteBufAllocator.java:231)
at io.netty.buffer.PooledByteBufAllocator.(PooledByteBufAllocator.java:220)
at io.netty.buffer.PooledByteBufAllocator.(PooledByteBufAllocator.java:206)
at io.netty.buffer.PooledByteBufAllocator.(PooledByteBufAllocator.java:188)
at io.netty.buffer.ByteBufUtil.(ByteBufUtil.java:87)
at io.netty.buffer.ByteBufAllocator.(ByteBufAllocator.java:24)
at io.netty.channel.DefaultChannelConfig.(DefaultChannelConfig.java:60)
at io.netty.channel.socket.DefaultServerSocketChannelConfig.(DefaultServerSocketChannelConfig.java:51)
...

--- a/common/src/main/java/io/netty/util/internal/CleanerJava24Linker.java
+++ b/common/src/main/java/io/netty/util/internal/CleanerJava24Linker.java
@@ -206,7 +206,9 @@ public class CleanerJava24Linker implements Cleaner {
     static long malloc(int capacity) {
         final long addr;
         try {
-            addr = (long) INVOKE_MALLOC.invokeExact((long) capacity);
+            // Always allocate at least 1 byte, to avoid relying on non-portable behavior
+            // of malloc(3) for zero size allocations.
+            addr = (long) INVOKE_MALLOC.invokeExact(Math.max(capacity, 1L));
         } catch (Throwable e) {
             throw new Error(e); // Should not happen.
         }

